extensions/amp-timeago/0.1/amp-timeago.js

```diff
@@ -0,0 +1,129 @@
+import {he} from './lib/timeago/locales/he';
+import {hu} from './lib/timeago/locales/hu';
+import {inBG} from './lib/timeago/locales/inBG';
+import {inHI} from './lib/timeago/locales/inHI';
+import {inID} from './lib/timeago/locales/inID';
+import {it} from './lib/timeago/locales/it';
+import {ja} from './lib/timeago/locales/ja';
+import {ko} from './lib/timeago/locales/ko';
+import {ml} from './lib/timeago/locales/ml';
+import {nbNO} from './lib/timeago/locales/nbNO';
+import {nl} from './lib/timeago/locales/nl';
+import {nnNO} from './lib/timeago/locales/nnNO';
+import {pl} from './lib/timeago/locales/pl';
+import {ptBR} from './lib/timeago/locales/ptBR';
+import {ro} from './lib/timeago/locales/ro';
+import {ru} from './lib/timeago/locales/ru';
+import {sv} from './lib/timeago/locales/sv';
+import {ta} from './lib/timeago/locales/ta';
+import {th} from './lib/timeago/locales/th';
+import {tr} from './lib/timeago/locales/tr';
+import {uk} from './lib/timeago/locales/uk';
+import {vi} from './lib/timeago/locales/vi';
+import {zhCN} from './lib/timeago/locales/zhCN';
+import {zhTW} from './lib/timeago/locales/zhTW';
+
+timeago.register('ar', ar);
+timeago.register('be', be);
+timeago.register('bg', bg);
+timeago.register('ca', ca);
+timeago.register('da', da);
+timeago.register('de', de);
+timeago.register('el', el);
+timeago.register('en', en);
+timeago.register('enShort', enShort);
+timeago.register('es', es);
+timeago.register('eu', eu);
+timeago.register('fi', fi);
+timeago.register('fr', fr);
+timeago.register('he', he);
+timeago.register('hu', hu);
+timeago.register('inBG', inBG);
+timeago.register('inHI', inHI);
+timeago.register('inID', inID);
+timeago.register('it', it);
+timeago.register('ja', ja);
+timeago.register('ko', ko);
+timeago.register('ml', ml);
+timeago.register('nbNO', nbNO);
+timeago.register('nl', nl);
+timeago.register('nnNO', nnNO);
+timeago.register('pl', pl);
+timeago.register('ptBR', ptBR);
+timeago.register('ro', ro);
+timeago.register('ru', ru);
+timeago.register('sv', sv);
+timeago.register('ta', ta);
+timeago.register('th', th);
+timeago.register('tr', tr);
+timeago.register('uk', uk);
+timeago.register('vi', vi);
+timeago.register('zhCN', zhCN);
+timeago.register('zhTW', zhTW);
+
+/** @private @const {string} */
+const DEFAULT_LOCALE_ = 'en';
+
+export class AmpTimeAgo extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {string} */
+    this.datetime_ = '';
+
+    /** @private {string} */
+    this.locale_ = '';
+
+    /** @private {string} */
+    this.title_ = '';
+  }
+
+  /** @override */
+  buildCallback() {
+    this.datetime_ = this.element.getAttribute('datetime');
+    this.locale_ = this.element.getAttribute('locale') || DEFAULT_LOCALE_;
+    this.title_ = this.element.textContent;
+
+    this.element.title = this.title_;
+    this.element.textContent = timeago(this.datetime_, this.locale_);
```

Comment:
To be honest I don't think enforcing `display: block` is going to work for a lot of use cases.

For example, on the BBC News website they display the article date/timeago inline next to the article location: 

![screen shot 2017-04-09 at 22 27 44](https://cloud.githubusercontent.com/assets/2844785/24841110/e8f9ced4-1d73-11e7-8ccd-688432c1d26a.png)

I think ideally we'd want to allow the component to use a flex-item layout. I'm wondering if there's a way of allowing this whilst ensuring the layout isn't redrawn?